src/core/a-entity.js

```diff
@@ -435,14 +435,28 @@ var proto = Object.create(ANode.prototype, {
       if (isDefault || isMixedIn) { return; }
 
       component = this.components[name];
+
+      if (!component) { return; }
+
+      // Wait for component to initialize.
+      if (!component.initialized) {
```

Comment:
Then the `removeComponent` call will never go through. A warning might be simpler though.